docs/spec/glossary.rst

```diff
@@ -0,0 +1,33 @@
+.. _`glossary`:
+
+Glossary
+========
+
+This section defines a few terms that may be used elsewhere in the specification.
+
+The definition of "MAY", "MUST", and "SHOULD", and "SHOULD NOT" are
+to be interpreted as described in :rfc:`2119`.
```

Comment:
I'd probably leave this at the top of the spec. These are not really typing-related glossary terms, but relate to how the spec is to be interpreted.

Reply:
Good point, done